transformer_engine/pytorch/ops/_common.py

```diff
@@ -98,8 +100,10 @@ def maybe_autocast_dtype(
     default_dtype: Optional[torch.dtype] = None,
 ) -> torch.dtype:
     """Get autocast dtype if enabled"""
-    if torch.is_autocast_enabled(device_type):
+    if get_torch_version() >= PkgVersion("2.4.3") and torch.is_autocast_enabled(device_type):
         return torch.get_autocast_dtype(device_type)
+    if torch.is_autocast_enabled():
+        return torch.get_autocast_gpu_dtype()
     return canonicalize_dtype(default_dtype)
```

Comment:
This will fail if PyT >= 2.4.3 and autocast is not enabled:
```suggestion
    if get_torch_version() >= PkgVersion("2.4.3"):
        if torch.is_autocast_enabled(device_type):
            return torch.get_autocast_dtype(device_type)
    else:
        if torch.is_autocast_enabled():
            return torch.get_autocast_gpu_dtype()
    return canonicalize_dtype(default_dtype)
```
Thoughts:
- Merging the `else`+`if` into an `elif` would be overclever and would obscure the intent. That said, we shouldn't fight Black or Pylint if they insist.
- If we modified the function to output a tuple, we don't need to bother with `Version`:
```python
if torch_version() >= (2, 4, 3):
```